config/common.php

```diff
@@ -17,17 +17,20 @@
 return [
     ValidatorInterface::class => Validator::class,
     RuleHandlerResolverInterface::class => SimpleRuleHandlerContainer::class,
-    'validator.categorySource' => static function (ContainerInterface $container) use ($params) {
-        $messageSource = $container->get('validator.messageSource');
-        $messageFormatter = $container->has(MessageFormatterInterface::class)
-            ? $container->get(MessageFormatterInterface::class)
-            : new SimpleMessageFormatter();
+    'validator.categorySource' => [
+        'definition' => static function (ContainerInterface $container) use ($params) {
+            $messageSource = $container->get('validator.messageSource');
+            $messageFormatter = $container->has(MessageFormatterInterface::class)
```

Comment:
`MessageFormatterInterface` don't should getting from container because it specific for each category.

Reply:
Done